bindings/nostr-sdk-ffi/src/transport/websocket.rs

```diff
@@ -0,0 +1,330 @@
+// Copyright (c) 2022-2023 Yuki Kishimoto
+// Copyright (c) 2023-2024 Rust Nostr Developers
+// Distributed under the MIT software license
+
+use std::fmt;
+use std::sync::Arc;
+use std::time::Duration;
+
+use async_wsocket::WsMessage;
+use nostr_sdk::async_utility::tokio::sync::{mpsc, Mutex};
+use nostr_sdk::pool::transport::websocket::{Sink, Stream};
+use uniffi::{Enum, Object};
+
+use crate::error::{NostrSdkError, Result};
+use crate::relay::ConnectionMode;
+
+#[derive(Debug, Enum)]
+pub enum WebSocketMessage {
+    Text(String),
+    Binary(Vec<u8>),
+    Ping(Vec<u8>),
+    Pong(Vec<u8>),
```

Comment:
Would be nice to have a Close(i32, String) or Close(Vec<u8>), similar to @spacecowboy 's suggestion.
```suggestion
    Pong(Vec<u8>),
    Close(i32, String) // or Close(Vec<u8>)
```